tasks/init.rb

```diff
@@ -126,7 +137,7 @@ def format_body(body, json)
   # overridden by a user.
   def format_headers(headers, json)
     default = json ? { 'Content-Type' => 'application/json' } : {}
-    default.merge(headers || {})
+    stringify_keys!(default.merge(headers || {}), deep: true)
```

Comment:
```suggestion
    default.merge(headers || {}).transform_keys(&:to_s)
```

The [task metadata](https://github.com/puppetlabs/puppetlabs-http_request/blob/main/tasks/init.json#L28) restricts the `headers` parameter to only accept Hashes with string keys and string values. Because of that, there's no need to recurse through the values, so we can just transform the keys with a built-in Hash method.